controller/WFCORE-433_git-persistence.adoc

```diff
@@ -0,0 +1,100 @@
+
+== Issue Metadata
+
+
+=== Issue
+
+*  {issue-base-url}/browse/WFCORE-433[WFCORE-433]
+
+=== Related Issues
+
+* {issue-base-url}/browse/EAP7-728[EAP7-728]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* mailto:msimka@redhat.com[Martin Simka]
+
+=== Affected Projects or Components
+
+WildFly Core kernel
+
+=== Other Interested Projects
+
+== Requirements
+
+=== Hard Requirements
+
+These items must be satisfied in order to have a satisfactory feature.
+
+* We are adding more parameters on the startup command line so that the user can configure the git repository (or its remote alias if the repository is already present) the branch (defaulting to master) or the tag to use. Something looking like:
+----
+    ./standalone --git-repository=file:///my_repo/test/.git --git-branch=my_branch
+----
+* Support for local only repository, initializing it on the first start-up:
+----
+    ./standalone --git-repository=local
+----
+* Support for remote repository pulling/cloning/pushing it before starting-up.
+* Automatic detection of a .git directory in the jboss.server.base.dir will activate this feature.
+* Support for the same features as the current persistence feature :
+** snapshots should be managed as tags
+** browsing history: list all the commits with their comments.
+** automatic persistence : committing each time the configuration files or content change.
+** restoring previous version: use the configuration state of a previous commit and revert to it.
+** try to use authentication information to store the user tagging or committing
+** publishing (aka pushing to a remote repository defined in the git configuration)
+* Adding support for comments when taking a snapshot: the comment will be used to create the tag name.
+* Extension to more elements of the configuration (aka managed contents) being versioned. When a managed content is no longer relevant it should be removed from the repository. +
+This supposes that the directory structure of the content directory and the configuration directory are the same as in the default configuration. Changing the jboss.server.base.dir is not a problem but we can’t manage several different places on a filesystem through Git.
+* Integration with Elytron for authentication : as we might need to authenticate on the remote repository instead of passing the username/password in clear text we could be using Elytron and pass the authentication credentials through a wildfly-config.xml.
+* Conflicts are to be resolved automatically. If that fails then it is up to the user to clean the mess using git itself as the tool is not meant to provide a full git implementation in jboss-cli.
+
+=== Nice-to-Have Requirements
+These items are not required but would be nice to have if possible:
+
+* Support for branch selection, but you can’t change the branch being used once the server has started-up.
+* Support for tag selection, but selecting a tag means that we are in a read-only mode.
```

Comment:
In current implementation new config changes can be committed (results in detached state). I think it's OK, just needs a note in the docs.